sound/soc/sof/sof-audio.c

```diff
@@ -253,7 +254,12 @@ static int sof_route_setup(struct snd_sof_dev *sdev, struct snd_soc_dapm_widget
 		return -EINVAL;
 	}
 
-	return ipc_tplg_ops->route_setup(sdev, sroute);
+	ret = ipc_tplg_ops->route_setup(sdev, sroute);
+	if (ret < 0)
+		return ret;
+
+	sroute->setup = true;
```

Comment:
Only moving the `sroute->setup = true;` here makes it IPC agnostic?
Should this be squashed to some previous patch?

Reply:
no its a fixup